lib/pester.rb

```diff
@@ -53,16 +53,10 @@ def self.retry_action(opts = {}, &block)
 
     opts[:max_attempts].times do |attempt_num|
       begin
-        result = yield block
-        return result
+        return yield(block)
       rescue => e
-        class_reraise = opts[:retry_error_classes] && !opts[:retry_error_classes].include?(e.class)
-        reraise_error = opts[:reraise_error_classes] && opts[:reraise_error_classes].include?(e.class)
-        message_reraise = opts[:message] && !e.message[opts[:message]]
-
-        if class_reraise || message_reraise || reraise_error
-          match_type = class_reraise ? 'class' : 'message'
-          opts[:logger].warn("Reraising exception from inside retry_action because provided #{match_type} was not matched.")
+        unless should_retry?(e, opts)
+          opts[:logger].warn("Reraising exception from inside retry_action.")
           raise
         end
```

Comment:
oops sorry hang on

On Wed, Apr 1, 2015 at 2:48 PM, Marc Bollinger <notifications@github.com>
wrote:

> In lib/pester.rb
> <https://github.com/lumoslabs/pester/pull/5#discussion_r27615408>:
>
> >        rescue => e
> > -        class_reraise = opts[:retry_error_classes] && !opts[:retry_error_classes].include?(e.class)
> > -        reraise_error = opts[:reraise_error_classes] && opts[:reraise_error_classes].include?(e.class)
> > -        message_reraise = opts[:message] && !e.message[opts[:message]]
> > -
> > -        if class_reraise || message_reraise || reraise_error
> > -          match_type = class_reraise ? 'class' : 'message'
> > -          opts[:logger].warn("Reraising exception from inside retry_action because provided #{match_type} was not matched.")
> > +        unless should_retry?(e, opts)
> > +          opts[:logger].warn("Reraising exception from inside retry_action.")
> >            raise
> >          end
>
> yeah, this reads much more nicely. replace those "s with '? that warning
> is still on in rubocop
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lumoslabs/pester/pull/5/files#r27615408>.
>